### PR TITLE
Fix Security and unit tests crashes

### PR DESCRIPTION
**Pull request by pestOO**
Security component:
* Add local variable for debugging
* Update includes
* Change DCHECK to RETURN
* Fixed Crash for SSL3

Unittests:
* Fix security tests
* Set correct certificate paths
* Fixed expectations
* Add checking decrypted string
* Fix crash in case no Context creation

### CODE REVIEW

**Comment by pestOO** [comment]
@AGaliuzov, @OHerasym, @VVeremjova, please review

**Comment by pestOO** [comment]
@AGaliuzov, @OHerasym, @Kozoriz , @VVeremjova, please review

File: src/components/security_manager/test/ssl_context_test.cc [hunk 1]
```diff
@@ -294,10 +310,10 @@ class SSLTestParam : public testing::TestWithParam<ProtocolAndCipher> {
       mock_crypto_manager_settings_;
   utils::SharedPtr<NiceMock<security_manager_test::MockCryptoManagerSettings> >
       mock_client_manager_settings_;
-  security_manager::CryptoManager* crypto_manager;
-  security_manager::CryptoManager* client_manager;
-  security_manager::SSLContext* server_ctx;
-  security_manager::SSLContext* client_ctx;
+  security_manager::CryptoManager* crypto_manager = NULL;
```
Comment (VVeremjova) [inline src/components/security_manager/test/ssl_context_test.cc]: @pestOO It is C++11, we should not initialize members at their declaration in class

Comment (pestOO) [inline src/components/security_manager/test/ssl_context_test.cc]: @VVeremjova, please see ccf4604caa91faaab7f7f5bb357d892b56ae57f3

**Comment by pestOO** [comment]
Commits squashed.
@AByzhynar, @AGritsevich, @anosach-luxoft, @AGaliuzov, please merge

**Comment by pestOO** [comment]
@LuxoftAKutsan, @AByzhynar, please support
Why manual format is not allowed by gcov?
https://travis-ci.org/smartdevicelink/sdl_core/builds/162784488#L544

File: src/components/security_manager/src/security_manager_impl.cc [hunk 2]
```diff
@@ -31,13 +31,17 @@
  */
 
 #include "security_manager/security_manager_impl.h"
-#include "security_manager/crypto_manager_impl.h"
-#include "protocol_handler/protocol_packet.h"
+
+#include <string.h>
```
Comment (AByzhynar) [inline src/components/security_manager/src/security_manager_impl.cc]: @pestOO May be `<cstring>`?

**Comment by AByzhynar** [comment]
@pestOO typo in 2d4c3bd7a46efcfd41ef847ae1b0e09e769eaa41 name

**Comment by AByzhynar** [comment]
@pestOO Fix coding style please. The problem is not in gcov. Utility clang-format failed the build as your code does not correspond c++03 std.

**Comment by pestOO** [comment]
@AByzhynar,
Please provide wrong `name` and link of build log where you find issue with c++03.

File: src/components/security_manager/src/security_manager_impl.cc (hunk 2 above)
Comment (pestOO) [inline src/components/security_manager/src/security_manager_impl.cc]: @AByzhynar, right, Will be fixed

**Comment by AByzhynar** [comment]
@pestOO  Use sdl_core/tools/infrastructure/check_style.sh locally in your branch. You have problems with spaces. Link to [build log](https://travis-ci.org/smartdevicelink/sdl_core/builds/162784488)

**Comment by pestOO** [comment]
@AByzhynar, thanks a lot!

**Comment by pestOO** [comment]
@AByzhynar, @AGritsevich, @anosach-luxoft, @AGaliuzov, please merge

**Pull request merged**